packages/wallet/ui/src/bridge-dapp.jsx

```diff
@@ -0,0 +1,195 @@
+// @ts-check
+import '@endo/eventual-send/shim';
+import './lockdown.js';
+
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { BridgeProtocol } from '@agoric/web-components';
+import { addOffer } from './store/Offers.js';
+import { loadDapp, upsertDapp, watchDapps } from './store/Dapps.js';
+import { OfferStatus } from './service/Offers';
+
+Error.stackTraceLimit = Infinity;
+
+/**
+ * @typedef {{ origin: string, chainId: string, address: string }} DappId
```

Comment:
By tuple I meant a fixed length array, but this works too. Though I think it should be `DappKey` b/c it's a composite key, not a primary id.

Reply:
Done